[DOC-6523] v23.1.0-rc.1 Release Notes (#16871)
cockroachdb/docs: _includes/releases/v23.1/v23.1.0-rc.1.md

## v23.1.0-rc.1

Release Date: May 2, 2023

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v23-1-0-rc-1-sql-language-changes">SQL language changes</h3>

> - The [`crdb_internal.transaction_contention_events` job](../v23.1/crdb-internal.html) now succeeds even if an error occurs getting the name of a non-SQL key. [#101871][#101871]

Reviewer: This is not a job afaik, it is a system table. 

Also, I think this belongs under Bug fixes.

```suggestion
- The [`crdb_internal.transaction_contention_events` table](../v23.1/crdb-internal.html) now generates even if an error occurs getting the name of a non-SQL key. [#101871][#101871]
```

Author: You're right, but I've put it under DB Console changes. When I was editing this, I included some info from a different item -- this is just a performance improvement.

> - The new SQL activity updater [job](../v23.1/admin-ui-jobs-page.html) updates the system tables `transaction_activity` and `statement_activity` to aggregate the top 500 of each of the 6 columns of the `statistics` tables. This improves the performance of the user interface. [#101995][#101995]

Reviewer: It's not clear what "SQL activity updater" is. Does it refer to `sql.stats.activity.persisted_rows.max` in the PR?

Author: No -- I didn't properly understand the change when I drafted this. I've made further edits now.

- Two views have been added to the [`crdb_internal` system catalog](../v23.1/crdb-internal.html): `crdb_internal.statement_activity` surfaces data in the persisted `system.statement_activity` table, and `crdb_internal.transaction_activity` surfaces the `system.transaction_activity` table. [#102047][#102047]

> - Span statistics are now unavailable on mixed-version clusters [#101878][#101878]

Reviewer: ```suggestion
- Span statistics are now unavailable on mixed-version clusters. [#101878][#101878]
```

<h3 id="v23-1-0-rc-1-db-console-changes">DB Console changes</h3>

- Timestamps are now displayed in DB Console in the timezone specified in the `ui.display_timezone` [cluster setting](../v23.1/cluster-settings.html), if set. Currently supported timezones are `Coordinated Universal Time` and `America/New_York`. [#102241][#102241]

<h3 id="v23-1-0-rc-1-bug-fixes">Bug fixes</h3>

> - Fixed a bug introduced prior to v22.1 where an `EXPORT` statement could be incorrectly evaluated and result in a node panic or incorrect query results if it had projection or rendering on top of the `EXPORT`, such as the statement`WITH cte AS (EXPORT INTO CSV 'nodelocal://1/export1/' FROM SELECT * FROM t) SELECT filename FROM cte;`. Only the presentation of the query result was affected, not the exported data. [#101806][#101806]

Reviewer: ```suggestion
- Fixed a bug introduced prior to v22.1 where an `EXPORT` statement could be incorrectly evaluated and result in a node panic or incorrect query results if it had projection or rendering on top of the `EXPORT`, such as the statement `WITH cte AS (EXPORT INTO CSV 'nodelocal://1/export1/' FROM SELECT * FROM t) SELECT filename FROM cte;`. Only the presentation of the query result was affected, not the exported data. [#101806][#101806]
```

> - The metric description of `rebalancing.readbytespersecond` and `rebalancing.writebytespersecond` now correctly reference bytes read and bytes written respectively. [#101710][#101710]

Reviewer: ```suggestion
- The descriptions of the `rebalancing.readbytespersecond` and `rebalancing.writebytespersecond` metrics now correctly reference bytes read and bytes written, respectively. [#101710][#101710]
```

> - Fixed a rare bug introduced prior to v22.1 where distributed plans shutdown could cause the graceful drain of a node to become stuck retrying forever.This bug leads to errors like `drain details: distSQL execution flows:` with a non-zero number of flows that is not reducing over a long period of time. [#101884][#101884]

Reviewer: Link to node shutdown doc?

> - Fixed a bug where a `RESTORE` operation with `skip_localities_check` could fail with errors if regions were missing on a cluster. [#101798][#101798]

Reviewer: Link to Restore doc.

> - Fixed a bug introduced in testing releases of v23.1 that could cause incorrect results for queries with `STRICT` user-defined functions. [#101951][#101951]

Reviewer: Link to UDF doc.

> - Fixed a rare bug that can cause keys to be unexpectedly deleted locally within a store by replica rebalancing during a write heavy workload. [#102166][#102166]

Reviewer: ```suggestion
- Fixed a rare bug that could cause keys to be unexpectedly deleted locally within a store by replica rebalancing during a write-heavy workload. [#102166][#102166]
```

> - Fixed a bug where a failed or cancelled `IMPORT` operation could leave some of the imported rows behind after it was cancelled, if the writing processes were slow enough to continue writing after the cleanup process started. [#102246][#102246]

Reviewer: Link to IMPORT doc.

> - Fixed a bug in the behavior of the `enforce_home_region` [session variable](../v23.1/set-vars) that may have allowed a hash join to be favored over a lookup join, or failed to error out remote accesses done by uniqueness checks for mutations on R`EGIONAL BY ROW` tables. Also, static erroring of some locality-optimized lookup joins is now handled dynamically during query execution. [#102287][#102287]

Reviewer: - Link to RBR doc.
- The part that starts "Also, static erroring" looks like it should be split into a separate note with the same PR link.

```suggestion
- Fixed a bug in the behavior of the `enforce_home_region` [session variable](../v23.1/set-vars) that may have allowed a hash join to be favored over a lookup join, or failed to error out remote accesses done by uniqueness checks for mutations on `REGIONAL BY ROW` tables. Also, static erroring of some locality-optimized lookup joins is now handled dynamically during query execution. [#102287][#102287]
```

> - Fixed a bug introduced in testing releases of v23.1 where a node could crash when evaluating a `COPY` command when the schema had `INT2` or `INT4` type. [#102307][#102307]

Reviewer: Link to COPY doc.

> - Fixes a bug where a backup of a key's revision history may not correctly be restored to the proper revision of the key if it is split across multiple sorted string tables. [#102342][#102342]

Reviewer: ```suggestion
- Fixed a bug where a backup of a key's revision history may not correctly be restored to the proper revision of the key if it is split across multiple sorted string tables. [#102342][#102342]
```

> - Fixed a bug introduced in testing releases of v23.1 where a user could be prevented from logging in or viewing or changing `GRANT`s if the cluster had a long period of inactivity. [#102488][#102488]

Reviewer: Link to GRANT doc.

> - Fixed a bug where a node that transfers a lease away and misses a liveness heartbeat could then erroneously update the closed timestamp during the stasis period if its liveness. This could lead to closed timestamp invariant violation and could cause the node to crash. In extreme cases, this could lead to inconsistencies in read-only queries. [#102602][#102602]

Reviewer: ```suggestion
- Fixed a bug where a node that transferred a lease away and missed a liveness heartbeat could then erroneously update the closed timestamp during the stasis period of its liveness. This could lead to closed timestamp invariant violation and could cause the node to crash. In extreme cases, this could lead to inconsistencies in read-only queries. [#102602][#102602]
```

<h3 id="v23-1-0-rc-1-performance-improvements">Performance improvements</h3>

> - SQL Activity dashboards now default to using a table optimized with the top data for the most used cases. Otherwise, they use persisted data if available, and in-memory data otherwise. [#102310][#102310]

Reviewer: Link to SQL Activity doc.

```suggestion
- SQL Activity dashboards now default to using a table optimized with the top data for the most -used cases. Otherwise, they use persisted data if available, and in-memory data otherwise. [#102310][#102310]
```

<h3 id="v23-1-0-rc-1-miscellaneous">Miscellaneous</h3>

- Two new statistics help to track the efficiency of snapshot transfers and to keep the number of failures due to system-level races as low as possible. `range.snapshots.recv-failed` shows the number of snapshot send attempts that are initiated but not accepted by the recipient. `range.snapshots.recv-unusable` shows the number of snapshots that were fully transmitted but not used. [#101835][#101835]

> <h4 id="v23-1-0-rc-1-changes-without-release-note-annotation">Changes without release note annotation</h4>

Reviewer: This doesn't belong here. The note below needs to be moved to the appropriate location.

> - Fixed per-connection memory leaks. [#101810][#101810]

Reviewer: This note also has no detail, so it needs to be rewritten.
